buildpacks/java/tests/build_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2021-2022 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package tests
+
+import (
+	"os"
+	"testing"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	. "github.com/onsi/gomega"
+
+	"kn-fn/java-function-buildpack/java"
+)
+
+const (
+	HTTPFuncTemplate = `from typing import Any
+
+def {{.Name}}({{range $var, $type := .Arguments}}{{$var}}:{{$type}}{{end}}):
+	return "{{.ReturnValue}}"`
+)
+
+func TestBuild(t *testing.T) {
+	spec.Run(t, "Build", testBuild, spec.Report(report.Terminal{}))
+}
+
+func testBuild(t *testing.T, when spec.G, it spec.S) {
+	var Expect = NewWithT(t).Expect
+
+	var (
+		build         java.Build
+		cleanupAppDir func()
+		context       libcnb.BuildContext
+	)
+
+	it.Before(func() {
+		build = java.Build{
+			Logger: NewLogger(),
+		}
+	})
+
+	it.After(func() {
+		cleanupAppDir()
+	})
+
+	when("#Build", func() {
+		var result libcnb.BuildResult
+
+		it.Before(func() {
+			var (
+				appDir string
+				err    error
+			)
+
+			Expect(os.Setenv("BP_FUNCTION", "some_module.some_function"))
+			appDir, cleanupAppDir = SetupTestDirectory(
+				WithFuncYaml(),
+				WithFunctionFile("some_module", "some_function", HTTPFuncTemplate),
+			)
+
+			context = makeBuildContext(
+				withBuildApplicationPath(appDir),
+				withDependencies([]map[string]any{
+					{"id": "invoker-deps", "version": "1.2.3"},
```

Comment:
`invoker-deps` is only for python, I believe. While I don't think it's hurting anything materially in the test, it should probably be removed to reduce confusion

Reply:
removed